Lib/test/test_imp.py

```diff
@@ -22,6 +22,13 @@ def requires_load_dynamic(meth):
     return unittest.skipIf(not hasattr(imp, 'load_dynamic'),
                            'imp.load_dynamic() required')(meth)
 
+def requires_create_dynamic(meth):
+    """Decorator to skip a test if not running under CPython or lacking
+    imp.create_dynamic()."""
+    meth = support.cpython_only(meth)
+    return unittest.skipIf(not hasattr(imp, 'create_dynamic'),
+                           'imp.create_dynamic() required')(meth)
```

Comment:
FWIW, as a reader it's easy to miss something when more than one thing is happening on a line.  It might be more clear to break this up:

    deco = unittest.skipIf(not hasattr(imp, 'create_dynamic'),
                                      'imp.create_dynamic() required')
    return deco(meth)

or even

    supported = hasattr(imp, 'create_dynamic')
    deco = unittest.skipIf(not supported, 'imp.create_dynamic() required')
    return deco(meth)